### DISCUSSION TITLE
horizontal lines (seem to be zero) in heat map

### DISCUSSION THREAD

**Discussion by yueyinqiu** [oxyplot/oxyplot#2065] [Q&A] [answered]
Hello! I'm using OxyPlot to draw the spectrums of some sound files.

Actually they are just a kind of `Heatmap`s. So I tried:

```csharp
Plot plot = new();
var heatmap = plot.Add.Heatmap(values);
plot.Add.ColorBar(heatmap);
plot.SavePng("test.png", 10000, 10000);
```

Of course I should also share the data here:

[values.txt](https://github.com/oxyplot/oxyplot/files/14445077/values.txt)

You can load the data file like:

```csharp
var lines = File.ReadAllLines("values.txt");
var length2 = lines[0].Split(',', StringSplitOptions.RemoveEmptyEntries).Length;

var values = new double[lines.Length, length2];
for(int i = 0; i < lines.Length; i++)
{
    var split = lines[i].Split(',', StringSplitOptions.RemoveEmptyEntries);
    for(int j = 0; j < split.Length; j++)
    {
        values[i, j] = double.Parse(split[j]);
    }
}
```

Then the previous program will generate a image like:

![test](https://github.com/oxyplot/oxyplot/assets/18749772/41ef372a-17c5-4638-a5f3-122134f01905)

It seems 'ugly', but in general it's what a spectrum should be. However these horizontal lines are not expected:

![image](https://github.com/oxyplot/oxyplot/assets/18749772/676747af-ed2a-49ff-affb-a387ed8783e6)

I have also tried to manually modify the `values` (this change is included in the previous data file). But it doesn't work. (`1` is the biggest value in the array. So there is expected to be a yellow line, but it's still blue.)

![image](https://github.com/oxyplot/oxyplot/assets/18749772/e451ed1e-77e7-4d2f-bd52-12c911b385ee)

So... have I did something wrong? Thanks for help in advance.

**Comment by yueyinqiu** [comment, marked as answer]
It's a problem in my other steps... And... `[511, 0]` will be drawn to the bottom left corner (`0, 0`) rather than `0, 511`

**Reply by VisualMelon** [reply]
You can specify the corners with the `X0`, `X1`, `Y0`, `Y1`, and `HeatMapCoordinateDefinition` properties

**Reply by yueyinqiu** [reply]
Yes. The problem did not happen during the drawing process but in the other steps. Thanks a lot!
